hammer/library/aws/elasticsearch.py

```diff
@@ -0,0 +1,356 @@
+            )["DomainStatus"]
+            if group_id in str(domain_details):
+                domains_list.append(ElasticSearchDomain_Details(
+                    domain_name=domain_name,
+                    domain_arn=domain_details["ARN"],
+                    vpc_id=domain_details["VPCOptions"]["VPCId"]
+                ))
+
+        return domains_list
+
+    @staticmethod
+    def put_domain_policy(es_client, domain_name, policy):
+        """
+        Replaces a policy on a domain. If the domain already has a policy, the one in this request completely replaces it.
+
+        :param es_client: Elasticsearch boto3 client
+        :param domain_name: Elasticsearch domain where to update policy on
+        :param policy: `dict` or `str` with policy. `Dict` will be transformed to string using pretty json.dumps().
+
+        :return: nothing
+        """
+        policy_json = jsonDumps(policy) if isinstance(policy, dict) else policy
+        es_client.update_elasticsearch_domain_config(
+            DomainName=domain_name,
+            AccessPolicies=policy_json,
+        )
+
+    @staticmethod
+    def retrieve_loggroup_arn(cw_client, domain_log_group_name):
+        """
+        This method used to retrieve cloud watch log group arn details if log group is available. If not, create a 
+         cloudwatch log group and returns arn of newly created log group
+        
+        :param cw_client: cloudwatch logs boto3 client
+        :param domain_log_group_name: Elasticsearch domain's log group name
+        :return: 
+        """
+        log_groups = cw_client.describe_log_groups()
+
+        log_group_arn = None
+        for log_group in log_groups["logGroups"]:
+            log_group_name = log_group["logGroupName"]
+            if log_group_name == domain_log_group_name:
+                log_group_arn = log_group["arn"]
+
+        if log_group_arn:
+            """
+            In order to successfully deliver the logs to your CloudWatch Logs log group, 
+            Amazon Elasticsearch Service (AES) will need access to two CloudWatch Logs API calls:
+                1. CreateLogStream: Create a CloudWatch Logs log stream for the log group you specified
+                2. PutLogEvents: Deliver CloudTrail events to the CloudWatch Logs log stream
+
+            Adding resource policy that grants above access.
+            """
+            policy_name = "AES-"+domain_log_group_name+"-Application-logs"
+            policy_doc = {}
+            statement = {}
+            principal = {}
+            action = []
+            principal["Service"] = "es.amazonaws.com"
+            action.append("logs:PutLogEvents")
+            action.append("logs:CreateLogStream")
+            statement["Effect"] = "Allow"
+            statement["Principal"] = principal
+            statement["Action"] = action
+            statement["Resource"] = log_group_arn
+
+            policy_doc["Statement"] = statement
+
+            cw_client.put_resource_policy(
```

Comment:
Please ensure Hammer has needed permissions to perform this action.

Reply:
Added required permissions.